src/site/_filters/patterns.js

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Finds a pattern by the patterns id in the paaterns data object.
+ * @param {string} id Pattern Id
+ * @param {Array} patterns Patterns data object
+ * @return {Object} Pattern object
```

Comment:
Can the pattern object get a typescript interface to be used to declare what the object should look like?

Reply:
Added in /lib/patterns.js